mne/io/egi/egimff.py

```diff
@@ -0,0 +1,411 @@
+                if isinstance(v, list):
+                    for k in v:
+                        if k not in event_codes:
+                            raise ValueError('Could find event named "%s"' % k)
+                elif v is not None:
+                    raise ValueError('`%s` must be None or of type list' % kk)
+
+            event_ids, = np.nonzero(np.unique(egi_events))
+            logger.info('    Synthesizing trigger channel "STI 014" ...')
+            logger.info('    Excluding events {%s} ...' %
+                        ", ".join([k for i, k in enumerate(event_codes)
+                                   if i not in include_]))
+            self.event_id = dict(zip([e for e in event_codes if e in
+                                      include_names], event_ids))
+        else:
+            # No events
+            self.event_id = None
+
+        info = _empty_info(egi_info['sfreq'])
+        info['buffer_size_sec'] = 1.  # reasonable default
+        my_time = datetime.datetime(
+            egi_info['year'], egi_info['month'], egi_info['day'],
+            egi_info['hour'], egi_info['minute'], egi_info['second'])
+        my_timestamp = time.mktime(my_time.timetuple())
+        info['meas_date'] = np.array([my_timestamp], dtype=np.float32)
+        ch_names = [channel_naming % (i + 1) for i in
+                    range(egi_info['n_channels'])]
+        ch_names.extend(list(egi_info['event_codes']))
+        if len(egi_events) > 0:
+            ch_names.append('STI 014')  # channel for combined events
+        ch_coil = FIFF.FIFFV_COIL_EEG
+        ch_kind = FIFF.FIFFV_EEG_CH
+        cals = np.concatenate(
+            [cals, np.repeat(1, len(event_codes) + 1 + len(misc) + len(eog))])
+        chs = _create_chs(ch_names, cals, ch_coil, ch_kind, eog, (), (), misc)
+        sti_ch_idx = [i for i, name in enumerate(ch_names) if
+                      name.startswith('STI') or name in event_codes]
+        for idx in sti_ch_idx:
+            chs[idx].update({'unit_mul': 0, 'cal': cals[idx],
+                             'kind': FIFF.FIFFV_STIM_CH,
+                             'coil_type': FIFF.FIFFV_COIL_NONE,
+                             'unit': FIFF.FIFF_UNIT_NONE})
+        info['chs'] = chs
+        info._update_redundant()
+        _check_update_montage(info, montage)
+        file_bin = os.path.join(input_fname, egi_info['eeg_fname'])
+        egi_info['egi_events'] = egi_events
+
+        self._filenames = [file_bin]
+        self._raw_extras = [egi_info]
+        super(RawMff, self).__init__(
+            info, preload=preload, orig_format=egi_info['orig_format'],
+            filenames=[file_bin], last_samps=[egi_info['n_samples'] - 1],
+            raw_extras=[egi_info], verbose=verbose)
+
+    def _read_segment_file(self, data, idx, fi, start, stop, cals, mult):
+        """Read a chunk of data."""
+        from ..utils import _mult_cal_one
+        dtype = '<f4'
```

Comment:
it seems like this should be read from the header somewhere but it's not currently (and the check above is broken)